plugins/fsm/src/__model__.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright © 2020, 2021 Anticrm Platform Contributors.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import core, { ArrayOf$, Builder, Class$, Mixin$, Prop, RefTo$ } from '@anticrm/model'
+import { CORE_CLASS_STRING, Ref, Class, Doc, MODEL_DOMAIN } from '@anticrm/core'
+import { Application, VDoc } from '@anticrm/domains'
+import { TDoc, TMixin } from '@anticrm/model/src/__model__'
+import { UX } from '@anticrm/presentation/src/__model__'
+import { IntlString } from '@anticrm/platform-i18n'
+
+import fsmPlugin, { FSM, Transition, State, WithFSM, WithState } from '.'
+import { ComponentExtension } from '@anticrm/presentation'
+import { AnyComponent } from '@anticrm/platform-ui'
+
+@UX('FSM' as IntlString)
+@Class$(fsmPlugin.class.FSM, core.class.Doc, MODEL_DOMAIN)
+export class TFSM extends TDoc implements FSM {
+  @UX('Name' as IntlString)
+  @Prop(CORE_CLASS_STRING)
+  name!: string
+
+  @RefTo$(core.class.Application)
+  application!: Ref<Application>
+
+  @UX('Name' as IntlString)
+  @ArrayOf$()
+  @RefTo$(fsmPlugin.class.Transition)
+  transitions!: Array<Ref<Transition>>
```

Comment:
Why we need an array? We could have Transition to have reference to FSM itself, and you could find all of transitions.